pkg/engine/jmespath/functions.go

```diff
@@ -686,7 +705,7 @@ func ifaceToString(iface interface{}) (string, error) {
 func validateArg(f string, arguments []interface{}, index int, expectedType reflect.Kind) (reflect.Value, error) {
 	arg := reflect.ValueOf(arguments[index])
 	if arg.Type().Kind() != expectedType {
-		return reflect.Value{}, fmt.Errorf(invalidArgumentTypeError, equalFold, index+1, expectedType.String())
+		return reflect.Value{}, fmt.Errorf(invalidArgumentTypeError, f, index+1, expectedType.String())
```

Comment:
Note - I noticed this small bug in the validation code where the output error message hardcoded the function name to `equalFold` instead of using `f` the name of the function that was actually called. Fixed this as well.